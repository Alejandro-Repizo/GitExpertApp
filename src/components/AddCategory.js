import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInput = (e) => {
		setInputValue(e.target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();

		if (inputValue.trim().length) {
			document.title = `GifExpertApp - ${inputValue}`;
			setCategories((categories) => [inputValue, ...categories]);
			setInputValue('');
		}
	};

	return (
		<form className='form' onSubmit={handleSubmit}>
			<input
				type='text'
				className='form__input'
				value={inputValue}
				onChange={handleInput}
			/>

			<button className='form__button' type='submit'>
				<i className='fas fa-search'></i>
			</button>
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
