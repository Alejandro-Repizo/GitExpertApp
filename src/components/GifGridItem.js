import React from 'react';

export const GifGridItem = ({ title, url }) => {
	return (
		<>
			<div className='card animate__animated animate__fadeIn'>
				<img src={url} alt={title} className='card__image' />
				<p className='card__title'>{title}</p>
			</div>
		</>
	);
};
