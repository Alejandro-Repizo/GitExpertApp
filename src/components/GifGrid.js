import React from 'react';
import PropTypes from 'prop-types';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);

	return (
		<div className='container'>
			<div className='wrapper wrapper--color'>
				<h3 className='container__title animate__animated animate__fadeIn'>
					<i className='fas fa-star icon icon--categories'></i>
					{category.charAt(0).toUpperCase() + category.slice(1)}
				</h3>

				{loading && (
					<h1 className='loader animate__animated animate__flash'>
						Loading...
					</h1>
				)}

				<div className='container__card'>
					{images.map((image) => (
						<GifGridItem key={image.id} {...image} />
					))}
				</div>
			</div>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};
