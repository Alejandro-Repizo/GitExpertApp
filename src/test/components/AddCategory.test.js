import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory/>', () => {
	const setCategories = jest.fn();
	let wrapper;

	beforeEach(() => {
		jest.clearAllMocks();
		wrapper = shallow(<AddCategory setCategories={setCategories} />);
	});

	test('<AddCategory/> se debe mostrar correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Debe de simular el evento change del input', () => {
		const input = wrapper.find('input');
		const value = 'Hola mundo!';
		input.simulate('change', { target: { value } });
	});

	test('No debe de lanzar la informaciÃ³n con onSubmit', () => {
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategories).not.toHaveBeenCalled();
	});

	test('Debe de llamar el setCategories y limpiar el input', () => {
		wrapper.find('input').simulate('change', { target: { value: 'category' } });
		wrapper.find('form').simulate('submit', { preventDefault() {} });
		expect(setCategories).toHaveBeenCalled();
		expect(wrapper.find('input').prop('value')).toBe('');
	});
});
