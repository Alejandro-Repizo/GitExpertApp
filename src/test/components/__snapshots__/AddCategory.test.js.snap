// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> <AddCategory/> se debe mostrar correctamente 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    className="form__input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    className="form__button"
    type="submit"
  >
    <i
      className="fas fa-search"
    />
  </button>
</form>
`;
